### PR TITLE
LITE-23303: Add translation sync command

### PR DESCRIPTION
**Pull request by qarlosh**
split in two commits. First one is refactoring, moving all synchronizer stats tracking and printing to a core helper class.
second one is adding the new translation sync command.

### CODE REVIEW

**Review by marcserrat — APPROVED** [review]

**Review by ffaraoneim — CHANGES_REQUESTED** [review]
I think we sould create a folder inside the plugins module for shared functions between plugins instead of put code that is not related with core features of the CLI in the core module.
IMO core features should be those related to generic command input/output, plugins management, configuration. 
Wdyt ?

**Comment by qarlosh** [comment]
> I think we sould create a folder inside the plugins module for shared functions between plugins instead of put code that is not related with core features of the CLI in the core module. IMO core features should be those related to generic command input/output, plugins management, configuration. Wdyt ?

sounds like a good idea, do I create a `connect/cli/plugins/shared` folder and move there shared functions between plugins?

**Review by d3rky — CHANGES_REQUESTED** [review]

File: connect/cli/plugins/translation/sync.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        translation_res = self._client.ns('localization').translations[translation_id]
+        progress = trange(0, max_counts, disable=self._silent, leave=False, bar_format=DEFAULT_BAR_FORMAT)
+        for _ in progress:
+            progress.set_description('Waiting for pending translation tasks')
+            sleep(wait_seconds or self.DEFAULT_WAIT_SECONDS)
+            pending_tasks = translation_res.tasks.filter(status__in=['processing', 'new']).count()
+            if not pending_tasks:  # pragma: no branch
+                break
+        else:
+            raise click.ClickException('Timeout waiting for pending translation tasks')
```
[the remark this answers is not in the document]
Comment (qarlosh) [reply connect/cli/plugins/translation/sync.py:189]: ok, I will check this!

**Review by d3rky — APPROVED** [review]

**Comment by qarlosh** [comment]
moved shared plugins functions to `connect/cli/plugins/shared`

**Review by ffaraoneim — APPROVED** [review]

**Pull request merged**